app/lib/actions/katello/foreman/content_update.rb

```diff
@@ -17,15 +18,18 @@ def plan(environment, content_view)
         def finalize
           environment  = ::Katello::KTEnvironment.find(input[:environment_id])
           content_view = ::Katello::ContentView.find(input[:content_view_id])
-          ::Katello::Foreman.update_puppet_environment(content_view, environment)
+          repository = ::Katello::Repository.find(input[:repository_id]) if input[:repository_id]
 
-          content_view.version(environment).repos(environment).each do |repo|
-            if distribution = repo.bootable_distribution
+          if content_view.default?
```

Comment:
Might as well check for repository here as well? `if repository && content_view.default?` or do you not want to want to process that else statement if there is no repository?